### PR TITLE
Adds a define for the max loop iterations

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

This is the highest number of iterations for a `for` loop before infinite loop detection kicks in.

## Changelog
:cl:
code: BYOND infinite loop detection is now documented.
/:cl:

### CODE REVIEW

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #define TEXT_EAST			"[EAST]"
 #define TEXT_WEST			"[WEST]"
 
+#define MAX_LOOP_ITERATIONS 1048574
```
Comment (ike709) [inline code/__DEFINES/misc.dm]:
```suggestion
//  The maximum number of times a "for()" loop can iterate before triggering infinite loop detection.
// +1 for "while()" loops, for some reason
#define MAX_LOOP_ITERATIONS 1048574
```

**Comment by PowerfulBacon** [comment]
I feel like it needs a clearer name, max loop iterations implies that loops won't work after this limit rather than the proc gets switched to background and the infinite loop detection kicks in.

File: code/__DEFINES/misc.dm [hunk 2]
```diff
@@ -9,6 +9,9 @@
 #define TEXT_EAST			"[EAST]"
 #define TEXT_WEST			"[WEST]"
 
+//  The maximum number of times a "for()" loop can iterate before triggering infinite loop detection.
+// +1 for "while()" loops, for some reason
+#define MAX_LOOP_ITERATIONS 1048574
```
Comment (ike709) [inline code/__DEFINES/misc.dm]:
```suggestion
//  The number of "for()" loop iterations at which infinite loop detection triggers
// +1 for "while()" loops, for some reason
#define INFINITE_LOOP_DETECTION_THRESHOLD 1048575
```

**Comment by ike709** [comment]
There

File: code/__DEFINES/misc.dm [hunk 3]
```diff
@@ -9,6 +9,9 @@
 #define TEXT_EAST			"[EAST]"
 #define TEXT_WEST			"[WEST]"
 
+//  The number of "for()" loop iterations at which infinite loop detection triggers
+// +1 for "while()" loops, for some reason
+#define INFINITE_LOOP_DETECTION_THRESHOLD 1048575
```
Comment (ike709) [inline code/__DEFINES/misc.dm]:
```suggestion
//  The highest number of "for()" loop iterations before infinite loop detection triggers
// +1 for "while()" loops, for some reason
#define INFINITE_LOOP_DETECTION_THRESHOLD 1048574
```

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
I thought this was already merged

**Pull request merged**